### PR TITLE
add 24.1 known limitations; reorganize existing KLs

### PR DESCRIPTION
**Pull request by taroface**
DOC-10100

[Preview](https://deploy-preview-18555--cockroachdb-docs.netlify.app/docs/dev/known-limitations#new-limitations-in-v24-1)

### CODE REVIEW

**Comment by taroface** [comment]
@DrewKimball @yuzefovich This looks like a bunch of changes, but all that needs to be reviewed is in the Preview link in the description, or `known-limitations.md`. Let me know if you have questions.

@michae2 Feel free to just review the addition to `read-committed-limitations.md`. I didn't include it in the "New" section since it was existing behavior (IIUC).

File: src/current/v24.1/known-limitations.md [hunk 1]
```diff
@@ -12,9 +12,24 @@ docs_area: releases
...
+
+### UDFs and stored procedures
+
+- Routines cannot be invoked with named arguments, e.g., `SELECT foo(a => 1, b => 2);` or `SELECT foo(b := 1, a := 2);`. [#122264](https://github.com/cockroachdb/cockroach/issues/122264)
+- Routines cannot be created if they reference temporary tables. [#121375](https://github.com/cockroachdb/cockroach/issues/121375)
+- Routines cannot be created with unnamed `INOUT` parameters. For example, `CREATE PROCEDURE p(INOUT INT) AS $$ BEGIN NULL; END; $$ LANGUAGE PLpgSQL;`. [#121251](https://github.com/cockroachdb/cockroach/issues/121251)
+- A `RECORD`-returning UDF cannot be created without a `RETURN` statement in the root block, which would restrict the wildcard type to a concrete one. [#122945](https://github.com/cockroachdb/cockroach/issues/122945)
+- UDFs cannot be created if they return fewer columns than declared. For example, `CREATE FUNCTION f(OUT sum INT, INOUT a INT, INOUT b INT) LANGUAGE SQL AS $$ SELECT (a + b, b); $$;`. [#121247](https://github.com/cockroachdb/cockroach/issues/121247)
```
Comment (yuzefovich) [inline src/current/v24.1/known-limitations.md:32]: nit: this limitation applies to both UDFs and stored procedures, so I would replace "UDFs" with "Routines".

Comment (taroface) [reply src/current/v24.1/known-limitations.md]: Fixed! Thanks.

**Review by yuzefovich — APPROVED** [review]
LGTM, thanks!

**Comment by taroface** [comment]
@mikeCRL PTAL at my includes here -- the idea is that the new KLs are added to:

1. the main Known Limitations page, directly.
2. the existing includes that appear on the respective (PL/pgSQL, UDF, stored procs) pages **as well as** under the "existing" heading on the Known Limitations page.

Within the includes, the new KLs for 24.1 are contained in a liquid block (which mentions v24.1 in an inline comment) that prevents them from being displayed on the Known Limitations page (i.e., appearing twice). The idea is that when we move the "new" KLs into the "existing" section later, we can:

1. delete everything under the "new" heading on the Known Limitations page.
2. move them out of the liquid blocks on the includes, thus causing them to appear in the Known Limitations "existing" section.

This sounds complicated but is hopefully intuitive once you look at it. Open to counter-suggestions & adjustments ofc.

**Review by mikeCRL — APPROVED** [review]
Really like the conditional-logic approach! Content LGTM, as well.

**Pull request merged**